### PR TITLE
ansible: install cmake3 on centos7_ppc64

### PR DESCRIPTION
**Pull request by sam-github**
Fixes: https://github.com/nodejs/build/issues/2321 as well as installing cmake3

### CODE REVIEW

**Comment by sam-github** [comment]
test-osuosl-centos7-ppc64--le-1 had its yum state all messed up, took some advice from https://wiki.centos.org/HowTos/CleaningUpDuplicatePackages, did some force reinstall of dupe packages, did some cleanups, did an update, cleaned up  transactions... the number of errors are going down, I think I'm making progress.

**Comment by sam-github** [comment]
Got a clean run on `test-osuosl-centos7-ppc64-le-1`.

**Comment by sam-github** [comment]
did the other 2 machines, looking good

**Comment by richardlau** [comment]
> 
> 
> test-osuosl-centos7-ppc64--le-1 had its yum state all messed up, took some advice from https://wiki.centos.org/HowTos/CleaningUpDuplicatePackages, did some force reinstall of dupe packages, did some cleanups, did an update, cleaned up transactions... the number of errors are going down, I think I'm making progress.

Jobs are failing to find `git`:
e.g. https://ci.nodejs.org/job/node-test-commit-plinux/nodes=centos7-ppcle/32703/console
```
Caused by: java.io.IOException: Cannot run program "git" (in directory "/home/iojs/build/workspace/node-test-commit-plinux/nodes/centos7-ppcle"): error=2, No such file or directory
```

File: ansible/roles/baselayout/vars/main.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ packages: {
   # partials/repo/centos7.yml for arm64
   centos7_arm64: ['git'], # git2u not available for aarch64 (yet)
   centos7_x64: ['devtoolset-6-libatomic-devel,git2u,centos-release-scl'],
-  centos7_ppc64: ['devtoolset-6-libatomic-devel,git,python3'],
+  centos7_ppc64: ['cmake3,devtoolset-6-libatomic-devel,git,python3'],
```
Comment (sam-github) [inline ansible/roles/baselayout/vars/main.yml:46]: @rvagg what is the syntax of the packages section? None of these packages are currently being installed, though I'd swear they used to be. I'm not having any luck figuring out what the OS+ARCH combination string should be in the packages table, or whether these are cumulative or not. I tried `centos7_ppc64le`.

Comment (AshCripps) [reply ansible/roles/baselayout/vars/main.yml:46]: When I logged into `centos7_ppc64le-2` earlier it had `cmake3` on it

Comment (sam-github) [reply ansible/roles/baselayout/vars/main.yml:46]: yes, but it doesn't get installed now, nor does git. I'm mystified.

Comment (richardlau) [reply ansible/roles/baselayout/vars/main.yml:46]: https://github.com/nodejs/build/issues/2321#issuecomment-624130373

File: ansible/roles/baselayout/tasks/main.yml [hunk 2]
```diff
@@ -66,8 +66,8 @@
   stat: path="{{ binary_dest }}"
   register: has_git
 
-- name: centos | remove existing git package
-  when: "(os == 'centos6' or os == 'centos7') and arch != 'arm64'"
+- name: centos6 | remove existing git package
+  when: "(os == 'centos6') and arch != 'arm64'"
```
Comment (sam-github) [inline ansible/roles/baselayout/tasks/main.yml:70]: AFAICT, this doesn't need to be done anymore. cc: @rvagg

File: ansible/roles/baselayout/tasks/main.yml [hunk 3]
```diff
@@ -79,6 +79,7 @@
     loop_var: package
   with_items:
     # ansible doesn't like empty lists
+    - "{{ packages[os+'_'+arch]|default('[]') }}"
```
Comment (sam-github) [inline ansible/roles/baselayout/tasks/main.yml:82]: this adds supports for _arch, thanks @richardlau

File: ansible/roles/baselayout/tasks/main.yml (hunk 2 above)
Comment (richardlau) [reply ansible/roles/baselayout/tasks/main.yml:70]: The brackets can be dropped here now the "or" clause is gone.

**Pull request merged**